### PR TITLE
HIVE-26790: update protobuf/grpc versions

### PR DESCRIPTION
**Pull request by henrib**
From [HIVE-26790]:

### What changes were proposed in this pull request?
Changing protobuf and grpc component versions

### Why are the changes needed?
Build on Apple M1

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Dependency version change, no new tests

### CODE REVIEW

**Review by cnauroth — APPROVED** [review]
+1 (non-binding)

The Jenkins CI failure appears to be something unrelated.

**Comment by dengzhhu653** [comment]
Hello @henrib, cloud you please rebase you changes on master? that would get a much quicker run.

**Comment by henrib** [comment]
Jenkins fails on building nightly due to hive-webhcat missing - haven't see where those should be built:

[2022-12-14T21:14:59.201Z] [INFO] Downloading from wonder: http://artifactory/artifactory/wonder/org/apache/hive/hcatalog/hive-webhcat/4.0.0-nightly-f95a16ad05-20221214_210643/hive-webhcat-4.0.0-nightly-f95a16ad05-20221214_210643.pom
[2022-12-14T21:15:00.577Z] [WARNING] The POM for org.apache.hive.hcatalog:hive-webhcat:jar:4.0.0-nightly-f95a16ad05-20221214_210643 is missing, no dependency information available
[2022-12-14T21:15:00.577Z] [INFO] Downloading from wonder: http://artifactory/artifactory/wonder/org/apache/hive/hcatalog/hive-webhcat-java-client/4.0.0-nightly-f95a16ad05-20221214_210643/hive-webhcat-java-client-4.0.0-nightly-f95a16ad05-20221214_210643.pom
[2022-12-14T21:15:01.954Z] [WARNING] The POM for org.apache.hive.hcatalog:hive-webhcat-java-client:jar:4.0.0-nightly-f95a16ad05-20221214_210643 is missing, no dependency information available
[2022-12-14T21:15:01.954Z] [INFO] Downloading from wonder: http://artifactory/artifactory/wonder/org/apache/hive/hcatalog/hive-webhcat/4.0.0-nightly-f95a16ad05-20221214_210643/hive-webhcat-4.0.0-nightly-f95a16ad05-20221214_210643.jar
[2022-12-14T21:15:01.954Z] [INFO] Downloading from wonder: http://artifactory/artifactory/wonder/org/apache/hive/hcatalog/hive-webhcat-java-client/4.0.0-nightly-f95a16ad05-20221214_210643/hive-webhcat-java-client-4.0.0-nightly-f95a16ad05-20221214_210643.jar

File: standalone-metastore/pom.xml [hunk 1]
```diff
@@ -89,7 +89,8 @@
     <log4j2.version>2.18.0</log4j2.version>
     <mockito-core.version>3.3.3</mockito-core.version>
     <orc.version>1.6.9</orc.version>
-    <protobuf.version>3.21.4</protobuf.version>
```
Comment (dengzhhu653) [inline standalone-metastore/pom.xml:92]: After some checking, I think we should also update the protobuf in the root pom.xml

Comment (nrg4878) [reply standalone-metastore/pom.xml:92]: @henrib looks like there is an unresolved comment. Has this been addressed?

Comment (henrib) [reply standalone-metastore/pom.xml:92]: Updated root pom.xml in new commit

**Review by dengzhhu653 — APPROVED** [review]

**Comment by deniskuzZ** [comment]
@henrib, have you tried to regenerate code with -Pprotobuf?

**Pull request merged**